### ISSUE TITLE
Retrieving detailed transit path results for more than 5,000 destinations (r5r)

### ISSUE THREAD

**Issue by mvpsaraiva** [conveyal/r5#750]
Hi. For `r5r`, we are implementing a new version of `travel_time_matrix()` that contains additional output information (such as `wait_time`, `transfer_time`, etc.), as discussed in [this](https://github.com/ipeaGIT/r5r/issues/194) issue. The idea for this feature started with [this comment](https://github.com/ipeaGIT/r5r/issues/144#issuecomment-789765078) from @ansoncfit.

However, `R5` currently has a hard limit of 5,000 destinations when retrieving path information (set in [this](https://github.com/conveyal/r5/blob/3387fd57daad40b4d1e3cd494cdf0ca98a4f21bc/src/main/java/com/conveyal/r5/analyst/cluster/PathResult.java#L65) line). So, `r5r` users can't get path information on very large study areas.

Perhaps this limit could be defined by a global static variable that we could change from within `r5r`?

**Comment by ansoncfit** [comment]
Thanks @mvpsaraiva. I think it would make sense to define that limit as a static variable (e.g. `MAX_DESTINATIONS_PER_ORIGIN`).

**Comment by abyrd** [comment]
Agreed, the limit is currently a literal constant "magic number" which does not conform to our usual code style anyway. We can make this a static non-final variable on the class so you can overwrite it.
